c-memory.go

```diff
@@ -0,0 +1,84 @@
+package mnemosyne
+
+import (
+	"context"
+	"math/rand"
+	"time"
+
+	"github.com/allegro/bigcache"
+	"github.com/sirupsen/logrus"
+)
+
+type inMemoryCache struct {
+	layerName          string
+	base               *bigcache.BigCache
+	amnesiaChance      int
+	compressionEnabled bool
+	cacheTTL           time.Duration
+	watcher            ITimer
+}
+
+func NewInMemoryCache(opts *CacheOpts, watcher ITimer) *inMemoryCache {
+	internalOpts := bigcache.Config{
+		Shards:             1024,
+		LifeWindow:         opts.cacheTTL,
+		MaxEntriesInWindow: 1100 * 10 * 60,
+		MaxEntrySize:       500,
+		Verbose:            false,
+		HardMaxCacheSize:   opts.memOpts.maxMem,
+		CleanWindow:        1 * time.Minute,
+	}
+	cacheInstance, err := bigcache.NewBigCache(internalOpts)
+	if err != nil {
+		logrus.Errorf("InMemCache Error: %v", err)
```

Comment:
@RRostami Consider using `%w` instead, as Go 1.13 [suggests](https://godoc.org/errors).

Reply:
`%w` only have meaning in `fmt.Errorf`. when printing to string with _logurs_ it doesn't mean anything.